blazor/sidebar/how-to/sidebar-in-the-dotnet8-application.md

```diff
@@ -0,0 +1,180 @@
+---
+layout: post
+title: Add Sidebar to MainLayout of the .NET 8 Blazor Web app | Syncfusion
+description: Learn here all about how to render the Sidebar component in the MainLayout page of the .NET 8 Blazor Web application and more.
+platform: Blazor
+control: Sidebar
+documentation: ug
+---
+
+<!-- markdownlint-disable MD009 -->
+
+# Render the Sidebar within the MainLayout of the .NET 8 Blazor Web app
+
+Integrate the Blazor Sidebar component into the `MainLayout.razor` page of the .NET 8 application. Next, include the `@rendermode InteractiveServer` directive in the `Routes.razor` page of the application.
+
+```cshtml
+
+@using Syncfusion.Blazor.Navigations
+@using Syncfusion.Blazor.Inputs
+<div class="page">
+    <div id="wrapper">
+        <div>
+            @*Initialize the Toolbar component*@
+            <SfToolbar>
+                <ToolbarEvents Clicked="@Toggle"></ToolbarEvents>
+                <ToolbarItems>
+                    <ToolbarItem PrefixIcon="e-tbar-menu-icon tb-icons" TooltipText="Menu"></ToolbarItem>
+                    <ToolbarItem>
+                        <Template>
+                            <div class="e-folder">
+                                <div class="e-folder-name">Navigation Pane</div>
+                            </div>
+                        </Template>
+                    </ToolbarItem>
+                </ToolbarItems>
+            </SfToolbar>
+        </div>
+        @*Initialize the Sidebar component*@
+        <SfSidebar HtmlAttributes="@HtmlAttribute" @ref="Sidebar" Width="290px" Target=".e-main-content" MediaQuery="(min-width:600px)" @bind-IsOpen="SidebarToggle">
+            <ChildContent>
+                <div class="main-menu">
+                    <div class="table-content">
+                        <SfTextBox Placeholder="Search..."></SfTextBox>
+                        <p class="main-menu-header">TABLE OF CONTENTS</p>
+                    </div>
+                    <div>
+                        <SfTreeView CssClass="main-treeview" ExpandOn="@Expand" TValue="TreeData">
+                            <TreeViewFieldsSettings Id="NodeId" NavigateUrl="NavigateUrl" Text="NodeText" IconCss="IconCss" DataSource="Treedata" HasChildren="HasChild" ParentID="Pid">
+                            </TreeViewFieldsSettings>                            
+                        </SfTreeView>
+                    </div>
+                </div>
+            </ChildContent>
+        </SfSidebar>
+        <!-- end of sidebar element -->
+        @*main-content declaration*@
+        <div class="main-content" id="main-text">
+            <div class="sidebar-content">
+                <div class="content px-4">
+                    @Body
+                </div>
+            </div>
+            <!--end of main content declaration -->
+        </div>
+    </div>
+</div>
+@code {
+    SfSidebar Sidebar;

```

Comment:
Sidebar reference it is not used here whether this is needed?

Reply:
Done.

Reply:
Done.